--- ../../vala-0.7.1/compiler/valacompiler.vala	2009-04-16 00:49:11.000000000 +0300
+++ gtkamlcompiler.vala	2009-05-04 19:48:12.000000000 +0300
@@ -1,5 +1,5 @@
-/* valacompiler.vala
- *
+/* gtkamlc.vala
+ * 
  * Copyright (C) 2006-2009  Jürg Billeter
  * Copyright (C) 1996-2002, 2004, 2005, 2006 Free Software Foundation, Inc.
  *
@@ -19,11 +19,14 @@
  *
  * Author:
  * 	Jürg Billeter <j@bitron.ch>
+ * Gtkaml adapted:
+ * 	Vlad Grecescu <b100dian@gmail.com>
  */
 
 using GLib;
+using Vala;
 
-class Vala.Compiler {
+class Gtkaml.Compiler {
 	static string basedir;
 	static string directory;
 	static bool version;
@@ -242,7 +245,7 @@
 		foreach (string source in sources) {
 			if (FileUtils.test (source, FileTest.EXISTS)) {
 				var rpath = realpath (source);
-				if (source.has_suffix (".vala") || source.has_suffix (".gs")) {
+				if (source.has_suffix (".vala") || source.has_suffix (".gs") || source.has_suffix (".gtkaml")) {
 					var source_file = new SourceFile (context, rpath);
 
 					// import the GLib namespace by default (namespace of backend-specific standard library)
@@ -271,12 +274,16 @@
 
 		var genie_parser = new Genie.Parser ();
 		genie_parser.parse (context);
+		
+		var gtkaml_parser = new MarkupParser ();
+		gtkaml_parser.parse (context);
+
 
 		if (context.report.get_errors () > 0) {
 			return quit ();
 		}
 		
-		var resolver = new SymbolResolver ();
+		var resolver = new MarkupResolver ();
 		resolver.resolve (context);
 		
 		if (context.report.get_errors () > 0) {
@@ -285,7 +292,6 @@
 
 		var analyzer = new SemanticAnalyzer ();
 		analyzer.analyze (context);
-
 		if (dump_tree != null) {
 			var code_writer = new CodeWriter (true);
 			code_writer.write_file (context, dump_tree);
@@ -454,7 +460,7 @@
 
 	static int main (string[] args) {
 		try {
-			var opt_context = new OptionContext ("- Vala Compiler");
+			var opt_context = new OptionContext ("- Vala Gtkaml Compiler");
 			opt_context.set_help_enabled (true);
 			opt_context.add_main_entries (options, null);
 			opt_context.parse (ref args);
@@ -465,7 +471,7 @@
 		}
 		
 		if (version) {
-			stdout.printf ("Vala %s\n", Config.PACKAGE_VERSION);
+			stdout.printf ("Gtkaml %s (based on Vala 0.7)\n", "0.4.0 the great rewrite");
 			return 0;
 		}
 		
